client/lib/domains/check-auth-code.js

```diff
@@ -6,12 +6,10 @@ export function checkAuthCode( domainName, authCode, onComplete ) {
 		return;
 	}
 
-	wpcom.undocumented().checkAuthCode( domainName, authCode, function ( serverError, result ) {
-		if ( serverError ) {
-			onComplete( { error: serverError.error, message: serverError.message } );
-			return;
-		}
-
-		onComplete( null, result );
-	} );
+	wpcom.req
+		.get( `/domains/${ encodeURIComponent( domainName ) }/inbound-transfer-check-auth-code`, {
+			auth_code: authCode,
+		} )
+		.then( ( data ) => onComplete( null, data ) )
+		.catch( ( error ) => onComplete( { error: error.error, message: error.message } ) );
```

Comment:
Seems like we're passing a new object with the same signature, so perhaps we could just pass `error`?

```suggestion
		.catch( ( error ) => onComplete( error ) );
```